src/main/webapp/dojump.jsp

```diff
@@ -30,7 +32,7 @@
             String url = request.getRequestURL().toString();
             url = url.substring(0, url.lastIndexOf('/') + 1);
             out.println("window.stop();");
-            out.println("location.replace('" + url + "error.jsp');");
+            out.println("location.replace('" + url + "error?errorMessage=" + errorMessage + "');");
```

Comment:
It looks like the basic idea behind the current approach is to introduce separate error pages + servlets and redirect to them wherever needed, but this has several disadvantages:

- You can see the error message in the URL
- All users will have all error messages they've ever gotten within their browser history
- You cannot use location.replace if your error occurs in the middle of a larger page if the page header has already been sent during the connection
- Bots/crawlers will think that the error page is a valid redirect from the source page
(and more)

So instead of creating separate servlets you should improve the error handling inside the default error page which is registered in the web.xml which should be called when exceptions are generated. Then  you can try to check for exception types + format the output per exception type.

I think there is even a method to define different error.jsp templates per exception type. There should be some tutorials on the web, here is an example:
https://www.digitalocean.com/community/tutorials/jsp-exception-handling-jsp-error-page

Reply:
Wie besprochen => ich schaue mal ob ich einen alternativen Lösungsvorschlag finde und melde mich nächste Woche dazu

Reply:
Alternativer Lösungsansatz siehe #267